### DISCUSSION TITLE
Rewrite Middleware in Route

### DISCUSSION THREAD

**Discussion by phhoef** [go-chi/chi#891] [Q&A] [answered]
I would like to rewrite the URL for a specific route.
I tried it like this, but it is not working
```go
	mux.Route("/endpoint", func(r chi.Router) {
              	r.Use(middleware.RewriteRoute)
		r.Route("/{source}", func(r chi.Router) {
			r.Get("/handler", myHandler)
		})
	})

```

Though, when moving the middleware to the outer scope it is working.
```go
	mux.Use(middleware.RewriteRoute)
	mux.Route("/endpoint", func(r chi.Router) {
		r.Route("/{source}", func(r chi.Router) {
			r.Get("/handler", myHandler)
		})
	})
```

Can someone please explain what's wrong with my idea?

**Comment by ousloob** [comment, marked as answer]
@phhoef 
When you use `mux.Route("/endpoint", func(r chi.Router) {...})`, Chi creates a subrouter that is isolated from the parent router mux. When you attach `middleware.RewriteRoute` inside this subrouter, it only affects the routes defined inside that block and does not alter the URL before it reaches the subrouter.
